### ISSUE TITLE
Add stubs to individual modules

### ISSUE THREAD

**Issue by abhi18av** [emilyncosta/mtbseq-nf#2]
- [ ] #4
- [ ] #3

**Comment by Mxrcon** [comment]
hey @abhi18av I'd like to contribute with this tasks, but I'm not confiden enough for working only in the main branch, I'll work on a branch to add the basic structure and them make a PR so you can squash and merge, what you think about it??

**Comment by abhi18av** [comment]
@Mxrcon  working on a `dev/develop` branch is a standard practice and you're on the right track. When many people are working on a repo, another practice is to use people/feature specific branch.

`davi/FEATURE_NAME` -> `dev` -> `main`.

Also, now I'd suggest you that whenever you want to pick up a task, you can assign to yourself (from the right hand menu on GH). 

Regarding the dev strategy, try to work on small PRs for early feedback. For example, `tbamend` and then raise a PR from `davi/tbamend` -> `develop` and then after some testing we can merge `develop` to `main`.

Does it make sense?

**Comment by Mxrcon** [comment]
Absolutely! Thanks for the clear instructions, I'll put them on pratice rn :+1:

**Comment by Mxrcon** [comment]
Also: I don't have permissions for assignment :(

**Comment by abhi18av** [comment]
Whoops!

Check now, you must have received an invitation (again).

**Comment by Mxrcon** [comment]
:smile: no problem, thanks for the invitation!

**Comment by abhi18av** [comment]
Btw, I have created a Github action on Tower and linked it to your branch as of now. Everytime  you’ll push on it, it’ll trigger the workflow. Notice the config I have used. Action shared on slack.

This is not a good practice to trigger for non-stable branches, but it’ll be good experience for you. Think of this a costlier (😅)  alternative for `Makefile` based dev cycle.

To make this work you’ll need to do the following

- Create a stub profile in the config
- Create and add the mock-data files in the repo i.e. /data/mock_data/xyz.fasta etc , remove the generator file.
- Use those files in the params via ${projectDir}/resources/mock_data/xyz pattern 


Be very careful with every push i.e. don’t push to GH unless you have tested enough locally.
